packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2643,41 +2654,27 @@ class _WidgetInspectorState extends State<WidgetInspector>
 
   final InspectorSelection selection;
 
-  /// Whether the inspector is in select mode.
-  ///
-  /// In select mode, pointer interactions trigger widget selection instead of
-  /// normal interactions. Otherwise the previously selected widget is
-  /// highlighted but the application can be interacted with normally.
-  bool isSelectMode = true;
-
   final GlobalKey _ignorePointerKey = GlobalKey();
 
   /// Distance from the edge of the bounding box for an element to consider
   /// as selecting the edge of the bounding box.
   static const double _edgeHitMargin = 2.0;
 
-  InspectorSelectionChangedCallback? _selectionChangedCallback;
   @override
   void initState() {
     super.initState();
 
-    _selectionChangedCallback = () {
-      setState(() {
-        // The [selection] property which the build method depends on has
-        // changed.
-      });
-    };
-    WidgetInspectorService.instance.selectionChangedCallback = _selectionChangedCallback;
+    WidgetInspectorService.instance._selectionChanged.addListener(_setState);
+    WidgetInspectorService.instance.isSelectMode.addListener(_setState);
   }
 
   @override
   void dispose() {
-    if (WidgetInspectorService.instance.selectionChangedCallback == _selectionChangedCallback) {
-      WidgetInspectorService.instance.selectionChangedCallback = null;
-    }
+    WidgetInspectorService.instance._selectionChanged.removeListener(_setState);
+    WidgetInspectorService.instance.isSelectMode.removeListener(_setState);
     super.dispose();
   }
-
+  void _setState() => setState((){});
```

Comment:
nit: its good practice to leave a comment inside the empty set describing why its empty (e.g. because WidgetInspectorService.instance.isSelectMode.value has changed and it is used in the built method we have to rebuild - or something like that.